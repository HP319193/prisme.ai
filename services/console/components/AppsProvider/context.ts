import { createContext, useContext } from 'react';

export interface AppsContext {
  apps: Map<string, Prismeai.App>;
  appInstances: Map<string, Prismeai.DetailedAppInstance[]>;
  getApps: ({
    text,
    page,
    limit,
    workspaceId,
  }: {
    text?: PrismeaiAPI.SearchApps.QueryParameters['text'];
    page?: PrismeaiAPI.SearchApps.QueryParameters['page'];
    limit?: PrismeaiAPI.SearchApps.QueryParameters['limit'];
    workspaceId?: PrismeaiAPI.SearchApps.QueryParameters['workspaceId'];
  }) => Promise<Prismeai.App[] | null>;
  getAppInstances: (
    workspaceId: PrismeaiAPI.ListAppInstances.PathParameters['workspaceId']
  ) => Promise<Prismeai.DetailedAppInstance[]>;
}

export const appsContext = createContext<AppsContext>({
  apps: new Map(),
  appInstances: new Map(),
  getApps: async () => ({} as any),
  getAppInstances: async () => ({} as any),
});

export const useApps = () => useContext(appsContext);

export default appsContext;
