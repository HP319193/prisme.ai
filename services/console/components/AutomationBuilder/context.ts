import { Schema } from '@prisme.ai/design-system';
import { createContext, useContext } from 'react';

export interface AutomationBuilderContext {
  automationId: string;
  addInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  removeInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  removeCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  editTrigger: () => void;
  editOutput: () => void;
  getApp: (
    instruction: string
  ) => {
    name: string;
    icon: string;
    instructionName: string;
  };
  instructionsSchemas: [
    string,
    Record<string, Schema & { description?: string }>,
    { icon: string }
  ][];
  getSchema: (name: string) => Schema;
}
export const automationBuilderContext = createContext<AutomationBuilderContext>(
  {
    automationId: '',
    addInstruction() {},
    removeInstruction() {},
    editInstruction() {},
    editCondition() {},
    removeCondition() {},
    editTrigger() {},
    editOutput() {},
    getApp() {
      return {
        name: '',
        icon: '',
        instructionName: '',
      };
    },
    instructionsSchemas: [],
    getSchema() {
      return {} as Schema;
    },
  }
);

export const useAutomationBuilder = () => useContext(automationBuilderContext);

export default automationBuilderContext;
