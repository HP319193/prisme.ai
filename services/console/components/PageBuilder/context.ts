import { createContext } from 'react';
import { Schema } from '@prisme.ai/design-system';
import { BlockInCatalog } from './useBlocks';
import { useContext } from '../../utils/useContext';

export type Block = NonNullable<Prismeai.Page['blocks']>[number];
export type BlocksWithKeys = Map<string, Block>;

export interface PageBuilderContext {
  value: BlocksWithKeys;
  addBlock: (position: number, blockName?: string) => void;
  setEditBlock: (blockId: string) => void;
  removeBlock: (key: string) => void;
  setBlockConfig: (key: string, config: any) => void;
  setBlockSchema: (blockId: string, schema: Schema | null) => void;
  catalog: BlockInCatalog[];
  blocksSchemas: Map<string, Schema | null>;
}

export const context = createContext<PageBuilderContext | undefined>(undefined);

export const usePageBuilder = () => useContext<PageBuilderContext>(context);
