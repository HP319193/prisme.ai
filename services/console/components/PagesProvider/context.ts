import { createContext, Dispatch, SetStateAction, useContext } from 'react';

type WorkspaceId = string;

export interface PagesContext {
  pages: Map<WorkspaceId, Set<Prismeai.Page>>;
  fetchPages: (workspaceId: NonNullable<Prismeai.Workspace['id']>) => void;
  setPages: Dispatch<SetStateAction<Map<string, Set<Prismeai.Page>>>>;
  createPage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  savePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  deletePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    pid: NonNullable<Prismeai.Page['id']>
  ) => Promise<Pick<Prismeai.Page, 'id'>>;
}

export const pagesContext = createContext<PagesContext>({
  pages: new Map(),
  setPages() {},
  fetchPages() {},
  async createPage() {
    return {} as Prismeai.Page;
  },
  async savePage() {
    return {} as Prismeai.Page;
  },
  async deletePage() {
    return {} as Prismeai.Page;
  },
});

export const usePages = () => useContext(pagesContext);

export default usePages;
