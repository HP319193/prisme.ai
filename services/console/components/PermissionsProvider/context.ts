import { createContext, useContext } from 'react';
import UserPermissions = Prismeai.UserPermissions;
import SubjectType = PrismeaiAPI.GetPermissions.Parameters.SubjectType;

export interface PermissionsContext {
  usersPermissions: Map<string, UserPermissions[]>;
  addUserPermissions: (
    subjectType: SubjectType,
    subjectId: string,
    permissions: UserPermissions
  ) => Promise<any>;
  // unshareWorkspaceToUser: (
  //   userEmail: string,
  //   workspaceId: string
  // ) => Promise<any>;
  getUsersPermissions: (
    subjectType: SubjectType,
    subjectId: string
  ) => Promise<UserPermissions[]>;
}

export const workspacesContext = createContext<PermissionsContext>({
  usersPermissions: new Map(),
  // unshareWorkspaceToUser: async () => ({} as any),
  addUserPermissions: async () => ({} as UserPermissions),
  getUsersPermissions: async () => [] as UserPermissions[],
});

export const usePermissions = () => useContext(workspacesContext);

export default workspacesContext;
