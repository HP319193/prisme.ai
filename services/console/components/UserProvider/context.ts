import { createContext, useContext } from 'react';
import { ApiError } from '@prisme.ai/sdk';

export enum OperationSuccess {
  emailSent = 'emailSent',
  passwordReset = 'passwordReset',
}

export interface ApiSuccess {
  type: OperationSuccess;
}
export interface UserContext<T = Prismeai.User | null> {
  user: T;
  loading: boolean;
  error?: ApiError;
  success?: ApiSuccess;
  signin: (email: string, password: string) => Promise<Prismeai.User | null>;
  signup: (
    email: string,
    password: string,
    firstName: string,
    lastName: string
  ) => Promise<Prismeai.User | null>;
  signout: (onServer?: boolean) => void;
  sendPasswordResetMail: (email: string, language: string) => Promise<any>;
  passwordReset: (token: string, password: string) => Promise<any>;
}

export const userContext = createContext<UserContext>({
  user: null,
  loading: false,
  signin: async () => null,
  signup: async () => null,
  signout() {},
  sendPasswordResetMail: async () => null,
  passwordReset: async () => null,
});

export function useUser(throwIfNotExist?: boolean): UserContext;
export function useUser(throwIfNotExist?: true): UserContext<Prismeai.User>;
export function useUser(throwIfNotExist?: boolean) {
  const context = useContext(userContext);

  if (throwIfNotExist && !context.user) {
    throw new Error();
  }

  return context;
}

export default userContext;
