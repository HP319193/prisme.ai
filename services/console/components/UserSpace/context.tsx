import { createContext } from 'react';
import { useContext } from '../../utils/useContext';

export interface UserSpaceConfig {
  /**
   * Main logo alternative
   */
  mainLogo?: {
    /**
     * Logo URL
     */
    url: string;
    /**
     * Attributes to apply on <img /> tag like alt or title
     */
    attrs?: object;
  };
  /**
   * Main URL on main logo link
   * @example: /product/ai-knowledge-chat/assistant?id=6474c0db33959b6283770367
   */
  mainUrl?: string;
  /**
   * The studio can act as a kiosk to keep the user in a URL template
   * @example: /product/ai-knowledge-chat/
   */
  kiosk?: string;
  /**
   * Define if Whats new popover is visible.
   * @default true
   */
  displayWhatsNew?: boolean;
  /**
   * Define if Help popover is visible.
   * @default true
   */
  displayHelp?: boolean;
  /**
   * Define if products popover is visible.
   * @default true
   */
  displayProducts?: boolean;

  /**
   * style object that will be injected inside various sections, allowing to cutomize theme colors ...
   */
  style?: {
    root?: object;
  };

  /**
   * Disable Builder
   */
  disableBuilder?: true;
}

interface UserSpaceProviderContext extends UserSpaceConfig {}
export const userSpaceContext = createContext<
  UserSpaceProviderContext | undefined
>(undefined);
export const useUserSpace = () =>
  useContext<UserSpaceProviderContext>(userSpaceContext);
export default userSpaceContext;
