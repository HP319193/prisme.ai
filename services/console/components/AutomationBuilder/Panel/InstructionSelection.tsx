import { useTranslation } from "next-i18next";
import Image from "next/image";
import { Button } from "primereact/button"
import { InputText } from "primereact/inputtext"
import { FC, useMemo, useState } from "react"
import { useAutomationBuilder } from "../context";

export interface InstructionSelectionProps {
  onSubmit: (key: string) => void;
}

export const InstructionSelection: FC<InstructionSelectionProps> = ({ onSubmit }) => {
  const { t } = useTranslation('workspaces');
  const { instructionsSchemas } = useAutomationBuilder();
  const [search, setSearch] = useState('');
  const filteredInstructions = useMemo(() => {
    return instructionsSchemas.reduce<[string, string, string[]][]>((prev, [name, list, more]) => {
      const matching = search ? Object.keys(list).filter(a => a.match(search)) : Object.keys(list)
      if (!matching) return prev;
      return [
        ...prev,
        [name, more.icon, matching]
      ];
    }, []);
  }, [instructionsSchemas, search])

  return (
    <>
      <div className="flex flex-1 align-items-stretch flex-column mb-4">
        <div>{t('automations.edit.select')}</div>
        <span className="p-input-icon-left p-input-icon-right">
          <i className="pi pi-search" />
          <InputText
            value={search}
            onChange={(e) => setSearch(e.target.value)}
            placeholder={t('automations.instruction.search')}
            autoFocus
          />
          <i><button onClick={() => setSearch('')} style={{ background: 'none', border: 0, color: 'inherit' }}><i className="pi pi-times" /></button></i>
        </span>
      </div>
      {filteredInstructions.map(([section, icon, instructionsInSection]) => (
        <div key={section} className="flex flex-1 flex-column mb-4">
          <div className="flex align-items-center">
            <div className="mr-2">
              <Image src={icon} width={16} height={16} alt={section} />
            </div>
            {section}
          </div>
          {instructionsInSection.map(key => (
            <Button
              key={key}
              onClick={() => onSubmit(key)}
              className="p-button-outlined my-1"
            >{key}</Button>
          ))}
        </div>
      ))}
    </>
  )
}

export default InstructionSelection
