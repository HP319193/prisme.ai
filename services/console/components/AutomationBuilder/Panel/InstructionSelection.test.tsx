import InstructionSelection from './InstructionSelection';
import renderer, { act } from 'react-test-renderer';
import { useAutomationBuilder } from '../context';
import { SearchInput } from '@prisme.ai/design-system';

jest.mock('../context', () => {
  const mock = {};
  return {
    useAutomationBuilder: () => mock,
  };
});

beforeEach(() => {
  useAutomationBuilder().instructionsSchemas = [
    [
      'logical',
      {
        emit: { search: 'emit' },
        wait: {
          search: 'wait',
        },
      },
      { icon: '/icon' },
    ],
    [
      'workspace',
      { kermit: { search: 'kermit' }, waitress: { search: 'waitress' } },
      { icon: '/icon' },
    ],
  ];
});

it('should render', () => {
  const onSubmit = jest.fn();
  const root = renderer.create(<InstructionSelection onSubmit={onSubmit} />);
  expect(root.toJSON()).toMatchSnapshot();
});

it('should filter', () => {
  const onSubmit = jest.fn();
  const root = renderer.create(<InstructionSelection onSubmit={onSubmit} />);

  act(() => {
    root.root
      .findByType(SearchInput)
      .props.onChange({ target: { value: 'ait' } });
  });

  expect(root.toJSON()).toMatchSnapshot();
});
