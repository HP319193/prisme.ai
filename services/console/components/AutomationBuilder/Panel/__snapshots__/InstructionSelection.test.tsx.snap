// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should filter 1`] = `
<div
  className="flex grow h-full flex-col overflow-auto"
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="logical"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="logical"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=%2Ficon&w=32&q=75"
        srcSet="/_next/image?url=%2Ficon&w=16&q=75 1x, /_next/image?url=%2Ficon&w=32&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  logical
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="workspace"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="workspace"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=%2Ficon&w=32&q=75"
        srcSet="/_next/image?url=%2Ficon&w=16&q=75 1x, /_next/image?url=%2Ficon&w=32&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  workspace
</div>
`;

exports[`should render 1`] = `
<div
  className="flex grow h-full flex-col overflow-auto"
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="logical"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="logical"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=%2Ficon&w=32&q=75"
        srcSet="/_next/image?url=%2Ficon&w=16&q=75 1x, /_next/image?url=%2Ficon&w=32&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  logical
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2716%27%20height=%2716%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="workspace"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="workspace"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=%2Ficon&w=32&q=75"
        srcSet="/_next/image?url=%2Ficon&w=16&q=75 1x, /_next/image?url=%2Ficon&w=32&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  workspace
</div>
`;
